doc/user/content/manage/user-management/rbac-newrole.md

```diff
@@ -0,0 +1,220 @@
+    assign role membership.
+
+    `CREATEDB` allows the role to create new databases.
+
+    `CREATECLUSTER` allows the role to create Materialize clusters. This
+    attribute is unique to the Materialize concept of clusters.
+
+## Create example objects
+
+Your `dev` role has the default system level permissions and needs object
+specific privileges. RBAC allows you to apply granular privileges to objects in the SQL hierarchy. Let's create some example objects in the system and determine what
+privileges the role needs.
+
+1. In your `psql` terminal, create a new example cluster to avoid impacting
+   other environments.
+
+   ```sql
+   CREATE CLUSTER devcluster;
+   ```
+
+1. Change into the example cluster.
+
+
+   ```sql
+   SET CLUSTER TO devcluster;
+   ```
+
+1. Create a new database, schema, and table in the cluster.
+
+   ```sql
+   CREATE DATABASE devdb;
+   ```
+
+   ```sql
+   CREATE SCHEMA devdb.schema;
+   ```
+
+   ```sql
+   CREATE TABLE d (a int, b text NOT NULL);
+   ```
+
+You just created a nested hierarchy of objects. Your schema object belongs to
+the database and the database is only accessible within the cluster. The next
+step is to grant privileges to your role based on the role needs.
+
+## Grant privileges to a role
+
+In this example, let's say your `dev` role needs the following permissions:
+
+* Read, write, and append privileges on the table
+* Update privileges on the schema
+* All available privileges on the database
+
+1. In your terminal, grant table-level privileges to the `dev` role.
+
+   ```sql
+   GRANT SELECT, UPDATE, INSERT ON d TO dev_role;
+   ```
+
+   Table objects have four available privileges - `read`, `write`, `insert`, and
+   `delete`. The `dev` role doesn't need `delete` permissions, so it is not
+   applied in the GRANT statement above.
+
+2. Grant schema privileges to the `dev` role.
+
+   ```sql
+   GRANT USAGE, CREATE ON SCHEMA devdb.schema TO dev_role;
+   ```
+
+   Schemas have `UPDATE` and `CREATE` privileges available to grant.
+
+3. Grant database privileges to the `dev` role.
+
+   ```sql
+   GRANT USAGE, CREATE ON DATABASE devdb TO dev_role;
+   ```
+
+4. Grant cluster privileges to the `dev` role.
+
+   ```sql
+   GRANT USAGE, CREATE ON CLUSTER devcluster TO dev_role;
+   ```
+
+   Materialize cluster privileges are unique to the Materialize RBAC structure.
+   To have access to the objects within a cluster, you must also have the same
+   level of access to the cluster itself.
+
+## Assign a role to a user
+
+The `dev` role now has the acceptable privileges it needs. Let's apply this role
+to a user in your Materialize organization.
+
+1. In your terminal, use the `GRANT` statement to apply a role to your new user.
+
+   ```sql
+   GRANT dev TO <new_user>;
+   ```
+
+1. To review the permissions a role has, you can view the object data.
+
+   ```sql
+   SELECT name, privileges::text FROM mz_tables WHERE name='t';
```

Comment:
I'm not sure if `privileges::text` is right here, what does an example output look like?

Reply:
Oh I see the output below. I didn't realize that this worked, that's neat.

Reply:
Ah that was actually meant to be a placeholder and I did not mark it well. This actually gives me `ERROR:  CAST does not support casting from mz_aclitem[] to text[]`